content/projects/kotlin/project-7/gridlayout-and-recyclerview/_index.md

```diff
@@ -0,0 +1,148 @@
+
+This app uses a simplified architecture with a UI controller, view model and LiveData, and a Room database to persist sleep data.
+
+![](49f975f1e5fe689.png)
+
+The sleep data is displayed in a RecyclerView. In this codelab, you change the app to use a GridLayout. The final screen will look like the screenshot below.
+
+![](b0abde98c5f99bf6.png)
+
+### Concept: Layouts and LayoutManagers
+
+In a previous project, when you added the RecyclerView to fragment_sleep_tracker.xml, you added a LinearLayoutManager without any customizations. This code displays the data as a vertical list.
+
+```
+app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
+```
+
+- LinearLayoutManager is the most common and straightforward layout manager for RecyclerView, and it supports both horizontal and vertical placement of child views. For example, you could use LinearLayoutManager to create a carousel of images that the user scrolls horizontally.
+
+### GridLayout
+
+Another common use case is needing to show a lot of data to the user, which you can do using GridLayout. The GridLayoutManager for RecyclerView lays out the data as a scrollable grid, as shown below.
+
+![](fcf0fc4b78f8650.png)
+
+From a design perspective, GridLayout is best for lists that can be represented as icons or images, such as lists within a photo browsing app. In the sleep-tracker app, you could show each night of sleep as a grid of large icons. This design would give the user an overview of their sleep quality at a glance.
+
+### How GridLayout lays out items
+
+GridLayout arranges items in a grid of rows and columns. Assuming vertical scrolling, by default, each item in a row takes up one "span." (In this case, one span is equivalent to the width of one column.)
+
+In the first two examples shown below, each row is made up of three spans. By default, the GridLayoutManager lays out each item in one span until the span count, which you specify. When it reaches the span count, it wraps to the next line.
+
+By default, each item takes up one span, but you can make an item wider by specifying how many spans it is to occupy. For example, the top item in the rightmost screen (shown below) takes up three spans.
+
+
+![](f51951e6ce5d8737.png)
+
+
+Tip: Span can mean either "row" or "column." With GridLayoutManager, you use spanCount to indicate how many columns or rows a grid has, and also how much grid space an item takes up horizontally or vertically.
+
+When you create a GridLayoutManager, you specify the orientation separately from the number of spans, and "span" is "direction-agnostic." In a (default) vertical configuration, "span" and "column" are equivalent.
+
+### Task: Implement GridLayout
+
+In this task, you take the RecyclerView that you finished in the last exercise and update it to display data using a GridLayoutManager. You can continue using the sleep-tracker app from the previous codelab, or you can download the RecyclerViewGridLayout-Starter app from GitHub.
```

Comment:
Does this imply that they should be using the same repo as before? Should the submission type not be `continue_repo`?

Reply:
I fell it's better to have an instruction that is completely new to each project and a new repo so that they don't have an excuse to not start a new project. That is unfortunately the type of people I have to deal with.